src/main/java/io/pravega/connectors/flink/FlinkPravegaOutputFormat.java

```diff
@@ -0,0 +1,279 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.pravega.connectors.flink;
+
+import io.pravega.client.ClientConfig;
+import io.pravega.client.ClientFactory;
+import io.pravega.client.stream.EventStreamWriter;
+import io.pravega.client.stream.EventWriterConfig;
+import io.pravega.client.stream.Serializer;
+import io.pravega.client.stream.Stream;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.flink.annotation.VisibleForTesting;
+import org.apache.flink.api.common.io.OutputFormat;
+import org.apache.flink.api.common.io.RichOutputFormat;
+import org.apache.flink.api.common.serialization.SerializationSchema;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.util.ExceptionUtils;
+import org.apache.flink.util.Preconditions;
+
+import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * A Flink {@link OutputFormat} that can be added as a sink to write into Pravega.
+ */
+@Slf4j
+public class FlinkPravegaOutputFormat<T> extends RichOutputFormat<T> {
+
+    private static final long serialVersionUID = 1L;
+
+    // The name of Pravega stream to write into.
+    private final String stream;
+
+    // The name of Pravega scope where the stream belongs to.
+    private final String scope;
+
+    // The supplied event serializer.
+    private final SerializationSchema<T> serializationSchema;
+
+    // The factory used to create Pravega clients; closing this will also close all Pravega connections.
+    private transient ClientFactory clientFactory;
+
+    // The Pravega client config.
+    private final ClientConfig clientConfig;
+
+    // The router used to partition events within a stream.
+    private final PravegaEventRouter<T> eventRouter;
+
+    // Pravega event writer instance.
+    private transient EventStreamWriter<T> pravegaWriter;
+
+    // Error which will be detected asynchronously and reported to Flink.
+    private final AtomicReference<Throwable> writeError;
+
+    // Used to track confirmation from all writes to ensure guaranteed writes.
```

Comment:
Should I interpret `guaranteed` write as durability?  Isn't durability and atomicity guaranteed via Pravega transactions? Can we use txns here?